src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -917,15 +920,32 @@ protected double computeCost(
 
         if (!isDiving)
         {
-            if (dY != 0 && !(ladder && parent.isLadder()) && !(Math.abs(dY) == 1 && below.is(BlockTags.STAIRS)))
+            boolean correctlyOnStairs = false;
+            if(Math.abs(dY) == 1 && below.getBlock() instanceof StairBlock && below.getValue(StairBlock.HALF) == Half.BOTTOM)
```

Comment:
Please reformat your code to ensure it's up to the defined code style (in this case, space between `if` and `(`

Reply:
> Please reformat your code to ensure it's up to the defined code style (in this case, space between `if` and `(`

Done.